pilot/pkg/model/config_test.go

```diff
@@ -186,19 +186,19 @@ func TestSubsetKey(t *testing.T) {
 			hostname: "hostname",
 			subset:   "subset",
 			port:     &model.Port{Name: "http", Port: 80, Protocol: model.ProtocolHTTP},
-			want:     "outbound|http|subset|hostname",
+			want:     "outbound.http.subset.hostname",
 		},
 		{
 			hostname: "hostname",
 			subset:   "subset",
 			port:     &model.Port{Port: 80, Protocol: model.ProtocolHTTP},
-			want:     "outbound||subset|hostname",
+			want:     "outbound..subset.hostname",
```

Comment:
Is this legal? Incidentally we do end up emitting such names in practice e.g. outbound.._default_.istio-sidecar-injector.istio-system.svc.cluster.local